packages/protobuf/src/reflect/reflect-check.ts

```diff
@@ -106,6 +121,94 @@ function checkSingular(
   return isReflectMessage(value, field.message);
 }
 
+type InvalidScalarValueErr = false | "invalid UTF8" | `${string} out of range`;
+
+function checkScalarValue(
```

Comment:
Previously, this function was exported from packages/protobuf/src/codegenv1/scalar.ts. This module seems to be a better fit for this function, since it is the only call-site. Same for function scalarTypeDescription.